gcovr/coverage.py

```diff
@@ -61,6 +61,47 @@ def update(self, other):
             self.throw = other.throw
 
 
+class DecisionCoverage(object):
+    r"""Represent coverage information about a decision.
+
+    Args:
+        count (int):
+            Number of times this decision was made.
+
+        uncheckable (bool):
+            The decision can't be checked because of instrumentation limits.
+    """
+
+    __slots__ = 'count', 'uncheckable'
+
+    def __init__(self, count, uncheckable):
+        # type: (int, Optional[bool]) -> None
+        assert count >= 0
+        self.count = count
+        self.uncheckable = uncheckable
+
+    @property
+    def is_covered(self):
+        # type: () -> bool
+        return self.count > 0
+
+    @property
+    def is_uncheckable(self):
+        # type: () -> bool
+        return self.uncheckable
+
+    def update(self, other):
+        # type: (DecisionCoverage) -> None
+        r"""Merge DecisionCoverage information"""
+        self.count += other.count
+        self.uncheckable = self.uncheckable or other.uncheckable
+
+    def update_count(self, count):
+        # type: (int) -> None
+        r"""Increase the counter by the given number"""
+        self.count += count
```

Comment:
All calls `a.update_count(b)` can be replaced with `a.count += b`, so this method isn't needed.